packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -316,13 +318,27 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
             color: game.backgroundColor(),
             child: LayoutBuilder(
               builder: (_, BoxConstraints constraints) {
-                game.onGameResize(constraints.biggest.toVector2());
+                final size = constraints.biggest.toVector2();
+                if (size.isZero()) {
+                  return widget.loadingBuilder?.call(context) ?? Container();
+                }
+                game.onGameResize(size);
                 return FutureBuilder(
                   future: loaderFuture,
                   builder: (_, snapshot) {
                     if (snapshot.hasError) {
                       final errorBuilder = widget.errorBuilder;
                       if (errorBuilder == null) {
+                        // @Since('2.16')
```

Comment:
can we just add dart>=2.16 to the pubspecs? otherwise I would just put out an issue with this code to tackle later, rather than commented code

Reply:
I'd love to! We have issue #1346 so that we won't forget to do that.